src/cpp/modules/manualEigenVector/ManualEigenVectorGMM.cpp

```diff
@@ -0,0 +1,78 @@
+// ManualEigenVectorGMM.cpp : Defines the exported functions for the DLL.
+#include "ManualEigenVectorGMM.h"
+#include "../../shared/gmm.h"
+#include "gmm_vector_d.h"
+#include "memory_size.h"
+
+#include <iostream>
+#include <iomanip>
+#include <memory>
+
+// This function must be called before any other function.
+void ManualEigenVectorGMM::prepare(GMMInput&& input)
+{
+    _input = input;
+    int Jcols = (_input.k * (_input.d + 1) * (_input.d + 2)) / 2;
+    _output = { 0,  std::vector<double>(Jcols) };
+
+    // If it is not enough memory in system to store all data that
+    // manualEigenVector need, then executing will interrupt
+    // to prevent system deadlock
+    unsigned long long int memory_size = get_memory_size();
+    // If memory_size != 0 then check memory size,
+    // else OS is undefined by get_memory_size()
+    // and we try to execute module
+    if (memory_size != 0) {
```

Comment:
If memory_size != 0 then compare it to the amount of memory we need, otherwise consider it indeterminable and continue the execution.